python/servo/build_commands.py

```diff
@@ -99,11 +126,30 @@ def build(self, build_type: BuildType, jobs=None, params=None, no_package=False,
             env["RUSTFLAGS"] = env.get("RUSTFLAGS", "") + " -Zsanitizer=address"
             opts += ["-Zbuild-std"]
             kwargs["target_override"] = target_triple
-            # TODO: Investigate sanitizers in C/C++ code:
-            # env.setdefault("CFLAGS", "")
-            # env.setdefault("CXXFLAGS", "")
-            # env["CFLAGS"] += " -fsanitize=address"
-            # env["CXXFLAGS"] += " -fsanitize=address"
+
+            # Note: We want to use the same clang/LLVM version as rustc.
+            rustc_llvm_version = get_rustc_llvm_version()
+            if rustc_llvm_version is None:
+                raise RuntimeError("Unable to determine necessary clang version for ASAN support")
+            llvm_major: int = rustc_llvm_version[0]
+            target_clang = f"clang-{llvm_major}"
+            target_cxx = f"clang++-{llvm_major}"
+            if shutil.which(target_clang) is None or shutil.which(target_cxx) is None:
+                raise RuntimeError(f"--with-asan requires `{target_clang}` and `{target_cxx}` to be in PATH")
+            env.setdefault("TARGET_CC", target_clang)
+            env.setdefault("TARGET_CXX", target_cxx)
+            # TODO: We should also parse the LLVM version from the clang compiler we chose.
+            # It's unclear if the major version being the same is sufficient.
+
+            # We need to use `TARGET_CFLAGS`, since we don't want to compile host dependencies with ASAN,
+            # since that causes issues when building build-scripts / proc macros.
+            env.setdefault("TARGET_CFLAGS", "")
+            env.setdefault("TARGET_CXXFLAGS", "")
```

Comment:
I know these are respected by cc-rs, but what about mozjs/mozangle? Either way this PR is still an improvement so I will not block it anymore.

Reply:
For mozjs the relevant logic is here: https://github.com/servo/mozjs/blob/728acdf3d4ce0604e9f75dd1d539dc6f291ccec7/mozjs-sys/build.rs#L1181 - I needed it before for cross-compiling to ohos. 

mozangle uses cc-rs, so it should be fine. I also should still have a clang wrapper script somewhere which I used before to check if certain options are passed through to all clang invocations.

Reply:
> mozangle uses cc-rs, so it should be fine

Right, I forgot and I was the one doing some refactorings there.

Great then we have everything covered, except maybe other external deps, but they are likely not (our) problem.